src/pycolmap/estimators/covariance.cc

```diff
@@ -42,8 +42,23 @@ void BindCovarianceEstimator(py::module& m) {
 
   using EstimatorBase = BundleAdjustmentCovarianceEstimatorBase;
   py::class_<EstimatorBase>(m, "BundleAdjustmentCovarianceEstimatorBase")
-      .def("has_pose", &EstimatorBase::HasPose, py::arg("image_id"))
+      .def(
+          "set_pose_blocks",
+          [](EstimatorBase& self, std::vector<py::array_t<double>>& pyarrays) {
+            std::vector<const double*> blocks;
+            for (auto it = pyarrays.begin(); it != pyarrays.end(); ++it) {
+              py::buffer_info info = it->request();
+              blocks.push_back((const double*)info.ptr);
+            }
```

Comment:
You could create a helper function to convert a `py::array_y<double>` to `std::vector<const double*>` as it is also used bellow.